### PR TITLE
[human_aware_navigation] Dynamically subscribing and unsubscribing from ppl perception topic.

### LINKED ISSUE #89: Make human_aware_navigation unsubscribe from ppl perception when not active
To not have ppl perception running when not necessary.

### PR DESCRIPTION
**Pull request by cdondrup**
Closes #89 

This prevents ppl perception from running when there is no need. Still has to be tested on robot.

### CODE REVIEW

File: strands_human_aware_navigation/scripts/human_aware_navigation.py [hunk 1]
```diff
@@ -116,7 +110,8 @@ def resetSpeed(self):
 
     def pedestrianCallback(self, pl):
         if not self._as.is_active():
-            rospy.logdebug("No active goal")
+            rospy.logdebug("No active goal. Unsubscribing.")
+            self.ppl_sub = None
```
Comment (marc-hanheide) [inline strands_human_aware_navigation/scripts/human_aware_navigation.py:115]: Is this enough? Does one not call [`Subscriber.unregister()`](http://docs.ros.org/indigo/api/rospy/html/rospy.topics.Subscriber-class.html#unregister)? I really don't know, just wondered...

Comment (cdondrup) [inline strands_human_aware_navigation/scripts/human_aware_navigation.py:115]: `Subscriber.unregister()` actually seems to cause some problems as we found out during the winter school and other people are struggling with on anwers.ros.org, e.g. [here](http://answers.ros.org/question/203735/unsubscribing-from-topics/)
Setting it to `None` lets the subscriber go out of scope and it therefore triggers the unsubscribing automatically.

Still have to test on the robot once the experiments are done.

**Comment by marc-hanheide** [comment]
OK, let's wait until you tested this, then.

**Comment by marc-hanheide** [comment]
Did you?

**Comment by cdondrup** [comment]
Is on my agenda for this week.

**Comment by cdondrup** [comment]
Actually @marc-hanheide, you were right once again. Setting it to None does not have to seem the effect I'd like. This does not explain why it worked in other applications though. Now calling `unregister` as well. Tested on robot.

this means that human aware navigation can be run and if there are no active goals it won't subscribe to the ppl perception and will not trigger it to start processing. Therefore all of this can be run in any case. I'll wait for Jenkins' approval and then merge.

**Comment by marc-hanheide** [comment]
I like :+1:

**Pull request merged**